### ISSUE TITLE
Batch Normalization layer gives significant difference between train and validation loss on the exact same data

### ISSUE THREAD

**Issue by izikgo** [fchollet/keras#7265]
Hi,
I have a pretty simple gist reproducing the problem: https://gist.github.com/izikgo/2579b8c26231d5c9a5a2c7d313860d33

In short, I get VERY different results between test and validation in a CNN with BN layers, even when I cancel `scale` and `center`, on the exact same data. The data is only a single batch of size 128 from the MNIST dataset.

Anyone knows if this is an acceptable behavior? I know that BN acts differently in train and inference, but the difference looks too big to me.

Thanks,
Izik

**Comment by srxdev0619** [comment]
I too am facing a similar issue, the distribution of activation of the same Conv layer is very different during training and inference on the same data. 

<img width="784" alt="screen shot 2017-07-07 at 2 51 46 pm" src="https://user-images.githubusercontent.com/9676573/27951880-08529702-6324-11e7-8d61-39346eece968.png">
 This is the distribution of activations during inference

<img width="890" alt="screen shot 2017-07-07 at 2 51 36 pm" src="https://user-images.githubusercontent.com/9676573/27951918-239b4a86-6324-11e7-98fa-077ba9ab7113.png">
This is the distribution of the activations during test.

The value of $\gamma$ is very close to 1 and the value of $\beta$ is very close to 0 for this particular layer.

**Comment by izikgo** [comment]
No after reading the code I understand why I'm getting these results. In training time there are two moving averages that are updated based on each batch - the mean and the variance. These values are supposed to approximate the population statistics. They are initialized to zero and one respectively, and then and each step multiplied by the momentum value (default is 0.99) and added the new value*0.01. At inference (test) time, the normalization uses these statistics. For this reason, it takes these values a little while to arrive at the "real" mean and variance of the data. If I lower the momentum for my specific example, the results makes much more sense...

**Comment by ysyyork** [comment]
Hi @izikgo , would you mind what value you set for momentum? I also came across this issue.

**Comment by guanzi0629** [comment]
@izikgo Thank you so much for you hint!!! I tried to reduce the momentum and solve it! I guess I need to read the paper to see the meaning of the momentum to understand the reason.
